streams/src/test/java/org/apache/kafka/streams/integration/MetricsReporterIntegrationTest.java

```diff
@@ -125,6 +126,8 @@ public void shouldBeAbleToProvideInitialMetricValueToMetricsReporter() {
             final Object initialMetricValue = METRIC_NAME_TO_INITIAL_VALUE.get(metricName.name());
             assertThat(initialMetricValue, notNullValue());
         });
+
+        kafkaStreams.close(Duration.ofSeconds(30));
```

Comment:
Should we use `try-with-resources` ? Do we need the timeout parameter (if yes, might still be better to use a try-catch with `finally` block?